### PR TITLE
NEW: Check for size of sample batch >= sgd_minibatch_size

### PR DESCRIPTION
**Pull request by kk-55**
## Why are these changes needed?

sgd_minibatch_size > size of sample batch causes an error since no minibatch can be generated

## Related issue number

https://discuss.ray.io/t/assertionerror-in-ppo-py-kl-is-none-learner-stats-of-at-least-one-policy-are-empty/2890/4?u=klausk55

Old and botched [PR](https://github.com/ray-project/ray/pull/17120#issuecomment-1027785186)

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — COMMENTED** [review]
looks reasonable. but can you revert all the diffs to cloudpickle please.

**Comment by kk-55** [comment]
> looks reasonable. but can you revert all the diffs to cloudpickle please.

@gjoliver done

File: rllib/utils/sgd.py [hunk 1]
```diff
@@ -48,6 +48,12 @@ def minibatches(samples: SampleBatch, sgd_minibatch_size: int, shuffle: bool = T
     if "state_in_0" not in samples and "state_out_0" not in samples:
         samples.shuffle()
 
+    assert (
+        samples.__len__ >= sgd_minibatch_size
```
Comment (gjoliver) [inline rllib/utils/sgd.py:52]: this should be __len__(). a lot of tests are failing.
also, why are you calling __len__() directly? shouldn't you use len(samples) instead?

Comment (kk-55) [reply rllib/utils/sgd.py]: Sorry @gjoliver, but I cannot remember the reason why I didn't use `len(samples)`. But I agree, it should also work with `len(samples)` and I've made an update.

**Review by gjoliver — APPROVED** [review]
thanks for the help.
@sven1977, can you help merge please?

**Pull request closed without merging**